dx-533/add-warning-to-property-type (#1262)
shopware/docs: guides/plugins/plugins/framework/data-handling/add-custom-complex-data.md

The entity class itself is a simple key-value object, like a struct, which contains as many properties as fields in the definition, ignoring the ID field, which is handled by the `EntityIdTrait`.

::: warning

> The properties of your entity class have to be at least `protected`, otherwise the data abstraction layer won't be able to set the values.

Author: ```suggestion
Properties of your entity class shouldn't be `private` so the data abstraction layer will be able to set the values.
```

Author: @AydinHassan yup, sounds better and is more clear. Good to go?

Reviewer: yep nice
